domain-models-runtime/src/main/resources/templates/db_scripts/metadata.ftl

```diff
@@ -22,33 +22,36 @@ CREATE OR REPLACE FUNCTION ${myuniversity}_${mymodule}.set_${table.tableName}_md
 RETURNS TRIGGER AS $$
 DECLARE
   createdDate timestamp WITH TIME ZONE;
-  createdBy text ;
+  createdBy text;
   updatedDate timestamp WITH TIME ZONE;
-  updatedBy text ;
-  injectedId text;
+  updatedBy text;
+
+  dateFormat text := 'YYYY-MM-DD"T"HH24:MI:SS.MSOF';
+  injectedMetadata jsonb;
+
 BEGIN
+  SET TIME ZONE 'UTC';
```

Comment:
This is used in a trigger, changing the time zone may have side effects for all SQL commands that run afterwards. Can you avoid setting the time zone?

Reply:
Currently it is the only way I found how to transform `timestamp` to `text` with UTC timezone. Actually it applies only to the current transaction. Or do you mean it can affect some SQL commands which is run in the current transaction after trigger execution?

Reply:
`to_char(updatedDate at time zone '+0000', dateFormat)` should do the trick.

Regarding `SET TIME ZONE`: "Once the surrounding transaction is committed, the effects will persist until the end of the session". https://www.postgresql.org/docs/current/sql-set.html

Reply:
It should not work in this case, `to_char` returns a string formatted with timezone set in configuration parameters, no matter which timezone the timestamp has.

As workaround timezone can be set to UTC before timestamp formatting and set back after.

Or it can be left at it is, but in this case timestamp is converted to string with default timezone offset like this `2019-01-01T02:00:00.000+02`

Reply:
@julianladisch also please take a loook at ffc6251f5c311b33f964e543fb3852f7eaa37807, it prevents potential issues if application and database have different timezones